### PR TITLE
Helper functions searchable documentation

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
https://sonarsource.github.io/SonarJS/typedoc/

### CODE REVIEW

**Comment by ilia-kebets-sonarsource** [comment]
TODOs:
- it might be nice to display the full signature in the search result, especially when the found term is a parameter, to make it easier to understand what was found
- I'd suggest to separate the different types of search result ('Types', vs 'Function', vs 'Class') if possible:
	Example: looking for 'isStringLiteral' I type 'literal' or 'string' in the search bar. The first result is the type 'StringLiteral'. It'd be nice to
	either clearly indicate it's a 'type' result (with the C, T, F, V icons on the side), or separate the types/variables/functions result with a clear separation bar or something. Maybe even put types at the end?
- minor UI remark -> if I'm in a element of the "Mocha" module (such as inside the 'isTestCase' function description), I clearly see on the right menu the vertical bar on the module hierarchy being highlighted, i.e. the "|" of " | SonarJS linter helper functions" and " | Mocha". If I'm in the description of an element from "SonarJS linter helper functions" (such as inside the 'flattenArgs' function description), there's no vertical bar highlight "|" next to the right "SonarJS linter helper functions" menu element.
- In some descriptions there are links to the defined types (example, in 'Mocha.extractTestCase' the 'TestCase' type is clickable to directly go on the page), and in others there aren't (example 'Express.attemptFindAppInjection' the 'RuleContext' is not clickable)

File: eslint-bridge/package.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
     "format": "prettier --write \"{src,tests,tools,profiling}/**/!(*.lint).ts\"",
     "compile": "tsc -b src tests profiling",
     "jar": "npm pack && mkdirp target/classes && mv eslint-bridge-1.0.0.tgz target/classes && bash tools/check-distribution-filepath-length.sh",
-    "new-rule": "ts-node tools/newRule.ts"
+    "new-rule": "ts-node tools/newRule.ts",
+    "td-plugin": "cd node_modules; ln -s ../typedoc/plugin my-plugin; cd ..",
```
Comment (ilia-kebets-sonarsource) [inline eslint-bridge/package.json:15]: move this into `eslint-bridge/typedoc`

Comment (ilia-kebets-sonarsource) [reply eslint-bridge/package.json:15]: and run it as part of `npm run td`

File: eslint-bridge/package.json [hunk 2]
```diff
@@ -42,7 +44,8 @@
     "mkdirp": "1.0.4",
     "prettier": "2.7.1",
     "ts-jest": "28.0.7",
-    "ts-node": "10.9.1"
+    "ts-node": "10.9.1",
+    "typedoc": "^0.23.18"
```
Comment (ilia-kebets-sonarsource) [inline eslint-bridge/package.json:48]: fix version

File: eslint-bridge/typedoc/searchable-parameters-plugin/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "Plugin for Typedoc that makes functions searchable by parameter types",
+  "main": "index.js",
+  "scripts": {
+    "setup": "node setup.js"
+  },
+  "keywords": ["typedoc-plugin"],
+  "author": "Ilia Kebets",
+  "license": "LGPL-3.0",
+  "dependencies": {
+    "typedoc": "^0.23.18"
```
Comment (ilia-kebets-sonarsource) [inline eslint-bridge/typedoc/searchable-parameters-plugin/package.json:13]: fix version

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
Tremendous work!

**Pull request merged**